apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl

```diff
@@ -191,6 +191,37 @@ t_monitor_sampler_format(_Config) ->
     [?assert(lists:member(SamplerName, SamplerKeys)) || SamplerName <- ?SAMPLER_LIST],
     ok.
 
+t_handle_old_monitor_data(_Config) ->
+    Now = erlang:system_time(second),
+    FakeOldData = maps:from_list(
+        lists:map(
+            fun(N) ->
+                Time = (Now - N) * 1000,
+                {Time, #{foo => 123}}
+            end,
+            lists:seq(0, 9)
+        )
+    ),
+
+    ok = meck:new(emqx, [passthrough, no_history]),
+    ok = meck:expect(emqx, running_nodes, fun() -> [node(), 'other@node'] end),
+    ok = meck:new(emqx_dashboard_proto_v1, [passthrough, no_history]),
+    ok = meck:expect(emqx_dashboard_proto_v1, do_sample, fun('other@node', _Time) ->
+        FakeOldData
+    end),
```

Comment:
Q: Won't this testcase risk becoming false positive once, for example, dashboard starts to use `emqx_dashboard_proto_v2`?

Reply:
I suspect the _history_ feature of `meck` could be helpful here?

Reply:
Yes, there is an updated variant where the call to stub is pinned.